portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -15,18 +15,79 @@
 package com.google.sps.servlets;
 
 import java.io.IOException;
+import java.util.*;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    List<String> messages;
```

Comment:
Would it be beneficial to initialize this when it is declared? I suspect there could be a NullPointException lurking in the code. For example if `toJson()` is called before this field is initialized  on line 52.
```suggestion
    List<String> messages = new ArrayList<>();
```

Reply:
I thought about this some more and I think this variable should be local to the doGet method. The servlet only gets instantiated once so messages would keep accumulating here until the server is shut down.